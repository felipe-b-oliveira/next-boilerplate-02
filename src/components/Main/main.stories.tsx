import type { Meta, StoryObj } from '@storybook/react'

import Main from './main'

const meta: Meta<typeof Main> = {
  title: 'Main',
  component: Main,
  args: {
    title: 'Storybook - React Avan√ßado',
    description: 'Storybook - TypeScript, ReactJS, NextJS e Styled Components'
  },
  argTypes: {
    title: Text,
    description: Text
  }
}

export default meta
type Story = StoryObj<typeof Main>

export const Basic: Story = {
  render: (args) => <Main {...args} />
}
