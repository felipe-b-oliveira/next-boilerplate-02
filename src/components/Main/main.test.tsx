import { render, screen } from '@testing-library/react'

import Main from './main'

/**
 * Error on 'toHaveStyle', the test seems to check for rgb colors istead of HEX '#COLOR_CODE'
 */
export const convertHexToRGBA = (hexCode: string) => {
  let hex = hexCode.replace('#', '')

  if (hex.length === 3) {
    hex = `${hex[0]}${hex[0]}${hex[1]}${hex[1]}${hex[2]}${hex[2]}`
  }

  const r = parseInt(hex.substring(0, 2), 16)
  const g = parseInt(hex.substring(2, 4), 16)
  const b = parseInt(hex.substring(4, 6), 16)

  return { r, g, b }
}

describe('<Main />', () => {
  it('should render the heading', () => {
    const { container } = render(<Main />)

    expect(screen.getByRole('heading', { name: /react avanÃ§ado/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': convertHexToRGBA('#06092b') })
  })
})
