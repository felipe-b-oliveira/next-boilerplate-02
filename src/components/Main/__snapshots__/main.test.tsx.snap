// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
<main
  class="mainstyles__Wrapper-ikEpiP hPgzrk"
>
  <img
    alt="Imagem de um átomo e React Avançado escrito ao lado."
    class="mainstyles__Logo-bTcZJv ctUzUz"
    src="/img/logo.svg"
  />
  <h1
    class="mainstyles__Title-llwSrw kZrEiE"
  >
    React Avançado
  </h1>
  <h2
    class="mainstyles__Description-joLNge FnQpa"
  >
    TypeScript, ReactJS, NextJS e Styled Components
  </h2>
  <img
    alt="Um desenvolvedor de frente para uma tela com código."
    class="mainstyles__Illustration-kobyzQ gpyygt"
    src="/img/hero-illustration.svg"
  />
</main>
`
